perception/sandbox/eyehandcal/src/eyehandcal/scripts/collect_data_and_cal.py

```diff
@@ -183,14 +188,17 @@ def extract_obs_data_std(data, camera_index):
     corner_data = detect_corners(data, target_idx=args.marker_id)
 
     num_of_camera=len(corner_data[0]['intrinsics'])
-    params=[]
+    CalibrationResult = namedtuple('CalibrationResult',
```

Comment:
It seems like factoring out stage1/stage2 out would make this script a little less massive, and separate the retry logic from the optimization problem.

Reply:
Agree. I want to add unit-test before the refactor, which should go to another PR to avoid a big PR which is too big to chew/review.